### PR TITLE
Single-client echo server

### PR DESCRIPTION
**Pull request by laurentqro**

### CODE REVIEW

File: src/main/java/EchoClient.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import java.net.Socket;
+
+public class EchoClient {
+    private ConsoleIO consoleIO;
+    private Socket echoSocket;
+
+    public static void main(String[] args) {
+        try {
+            String hostName = "localhost";
+            int portNumber = 8000;
```
Comment (fabientownsend) [inline src/main/java/EchoClient.java:11]: What if you want to run your echoServer on the port 8001? I think you would have to recompile and rerun your app
You can avoid that by writing that on a config file that your app would read when you run it 👍

File: src/main/java/EchoClient.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    public EchoClient(ConsoleIO consoleIO, Socket echoSocket) {
+        this.consoleIO = consoleIO;
+        this.echoSocket = echoSocket;
+    }
+
+    public void start() {
+        try {
+            BufferedReader in = new BufferedReader(new InputStreamReader(echoSocket.getInputStream()));
+            PrintWriter out = new PrintWriter(echoSocket.getOutputStream(), true);
```
Comment (fabientownsend) [inline src/main/java/EchoClient.java:29]: Can `new BufferedReader` and `new PrintWriter` raise an exception? If it's not the case, I would put it out of the try

File: src/main/java/EchoClient.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+            String userInput;
```
Comment (fabientownsend) [inline src/main/java/EchoClient.java:31]: See comment above. I would use the try {} as a way to communicate that a specific part of the code can crash and for this reason remove anything that would not raise any exception

File: src/main/java/ConsoleIO.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import java.io.*;
+
+public class ConsoleIO {
+    private BufferedReader inputStream;
+    private OutputStream outputStream;
+    private PrintStream printStream;
+
+    ConsoleIO(InputStream inputStream, OutputStream outputStream) {
+        this.inputStream = new BufferedReader(new InputStreamReader(inputStream));
+        this.outputStream = outputStream;
```
Comment (RabeaGleissner) [inline src/main/java/ConsoleIO.java:10]: Do you need this line? Maybe I'm missing something here but I don't see that you're using `this.outputStream` anywhere.

File: src/test/java/EchoClientTest.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.net.Socket;
+
+import static org.junit.Assert.*;
+
+public class EchoClientTest {
+    @Test
+    public void fromStdInAndBack() {
```
Comment (RabeaGleissner) [inline src/test/java/EchoClientTest.java:11]: I'd make the test name a bit more descriptive. Something like `sent message is echoed back` maybe?

File: src/main/java/EchoClient.java (hunk 3 above)
Comment (laurentqro) [reply src/main/java/EchoClient.java:31]: `echoSocket.getInputStream()` and `echoSocket.getOutputStream()` raise an `IOException`.

**Comment by laurentqro** [comment]
@RabeaGleissner @fabientownsend thanks for your feedback, very much appreciated!

**Comment by RabeaGleissner** [comment]
@laurentqro No problem! Are you going to implement our feedback?!

**Comment by laurentqro** [comment]
@RabeaGleissner yep, you'll find it here: https://github.com/laurentqro/echoServer/pull/3

**Pull request merged**